gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -800,6 +802,14 @@ public EnumTypeAdapter(Class<T> classOfT) {
             for (String alternate : annotation.alternate()) {
               nameToConstant.put(alternate, constant);
             }
+            String regex = annotation.pattern();
+            if (regex != "") {
```

Comment:
nit: almost always want `.equals()` on strings. (These will be interned, but it’s still awkward)